_schemaV2.graphql

```diff
@@ -6336,6 +6336,29 @@ type FairExhibitorsGroup {
   letter: String
 }
 
+type FairOrganizer {
+  about: String
+  fairsConnection(
+    after: String
+    before: String
+    first: Int
+    last: Int
+  ): FairConnection
+
+  # A globally unique ID.
+  id: ID!
+
+  # A type-specific ID likely used as a database ID.
+  internalID: ID!
+  name: String
+  profile: Profile
+  profileID: ID
+
+  # A slug ID.
+  slug: ID!
+  website: String
+}
+
```

Comment:
Looks great! But according to [this figma spec](https://www.figma.com/file/aM3b5sodn04vMTigbQXACi/Fairs-2021?node-id=3139%3A5588) we want to get cover image, icon image, startAt/endAt dates, and social links that are missing here.

Does that make sense? I'm just a bit unclear on what exactly do we show on fair organizer page. Seems like there's info about some closest/main event of the organizer's in overview block cause it has start/end dates, _Art Paris: Art Fair_ writing on icon image. Or fair organizer is an event itself?

So my question is should we get above mentioned data from `fairOrganizer` endpoint or from `fairsConnection`?
In the first case it seems that the schema is not complete

Reply:
@anastasiapyzhik those are really good questions, thanks for flagging!

I did some digging, and here's what I found:
- Cover image & icon image: these can be accessed via the `profile` field included on `fairOrganizer`:
```graphql
{
  fairOrganizer(id:"art-paris") {
    profile {
      icon {
        imageVersions
        url(version:"square") 
        # note that we store several different image types, listed with imageVersions; when we
        # implement these pages we should make sure we're using the right size for the situation!
      }
      image {
        imageVersions
        url(version:"large")
      }
    }
  }
}
```
- startAt/endAt: our [current implementation](https://github.com/artsy/force/blob/ac64ba6786e85b8231bfb5e5f2e419fd3bd3b5be/src/desktop/apps/fair_organizer/client/index.coffee#L46-L54) gets these dates from an upcoming fair if one exists (e.g. for Art Paris, those dates are from `art-paris-2021`). The fair organizer model itself does not store any information about editions of the fair. So we should use `fairsConnection` to fetch these dates going forward.
- Social links: OK, this is a really weird one 😅 we don't currently make these available via the API as individual links - they're simply part of a fair organizer's `about` text. For example, here's the "about" text that is returned from our API for Art Paris:
```
"Art Paris, the leading spring event for modern and contemporary art, supports and celebrates the French art scene and invites visitors to discover the Spanish and Portuguese art from the 1950s to the present day, from modern masters to contemporary artists.\r\n\r\nIn the 20 years since its first edition, Art Paris has become Paris’s must-see spring fair for modern and contemporary art, once again bringing together more than 150 galleries from over 20 different countries with a total of almost 1,000 artists.\r\n\r\nIn 2021, Art Paris will become the first art event to move into the ephemeral Grand Palais which will be installed on the Champ de Mars in Paris, while renovation work is carried out on the historic Parisian monument.\r\n\r\n**Follow the Fair**\r\n[Facebook](https://www.facebook.com/ArtParisArtFair/) | [Instagram](https://www.instagram.com/artparisartfair/) | [Twitter](https://twitter.com/ArtParisArtFair)"
```
So in this case, it seems like we will want to add new fields to our `FairOrganizer` model in Gravity to support storing these links in a structured way. I've ticketed that work [here](https://artsyproduct.atlassian.net/browse/FX-3141).

Also, pasting these links so you can quickly see what API responses for different fields look like in case it's useful:
Fair profile: https://stagingapi.artsy.net/api/v1/profiles?id[]=the-armory-show
Fairs filtered by fair organizer: https://stagingapi.artsy.net/api/v1/fairs?fair_organizer_id=the-armory-show
Fair organizer: https://stagingapi.artsy.net/api/v1/fair_organizer/the-armory-show

I hope that helps! Let me know if you have other questions :)